nodejs/rest-api/integration.tests/rest.api.contracts.test.js

```diff
@@ -0,0 +1,110 @@
+
+
+const assertContracts = async (isMainnet) => {
+    const contracts = await restApiHandler.getContracts(isMainnet);
+    expect(contracts).to.have.property('status').equals(200);
+    expect(contracts).to.have.property('data');
+    expect(contracts.data).to.have.property('contracts');
+    expect(contracts.data.contracts).to.be.an('array');
+    expect(contracts.data.contracts).to.have.lengthOf.at.least(1)
+
+    //Testing for the properties on the first contract
+
+    const firstContract = contracts.data.contracts[0];
+    expect(firstContract).to.have.property('contract_id').that.is.a('string');
+    expect(firstContract).to.have.property('created_timestamp').that.is.a('string');
+    expect(firstContract).to.have.property('expiration_timestamp').that.is.a('string');
+    expect(firstContract).to.have.property('file_id');
+    expect(firstContract).to.have.property('evm_address').that.is.a('string');
+}
+
+
+const assertContractById = async (isMainnet) => {
+    const contractResponse = await restApiHandler.getContractById(testnetContract.id ,isMainnet);
+    const contract = contractResponse.data;
+    expect(contract).to.have.property('contract_id').that.is.a('string');
+    expect(contract).to.have.property('created_timestamp').that.is.a('string');
+    expect(contract).to.have.property('expiration_timestamp').that.is.a('string');
+    expect(contract).to.have.property('file_id');
+    expect(contract).to.have.property('evm_address').that.is.a('string');
+}
+
+const assertContractResultsById = async (isMainnet) => {
+    const contractResponse = await restApiHandler.getContractResultsById(testnetContract.id ,isMainnet);
+    const contract = contractResponse.data;
+    expect(contract).to.have.property('results');
+}
+
+const assertContractResultLogsById = async (isMainnet) => {
+    const contractResponse = await restApiHandler.getContractResultLogsById(testnetContract.id ,isMainnet);
+    const contract = contractResponse.data;
+    expect(contract).to.have.property('logs');
+}
+
+
+const assertContractEvmCalls = async (isMainnet) => {
+    // Arrange.
+    const provider = new ethers.providers.JsonRpcProvider(urlHandler.getJsonRpcTestnet());   
+    let callAbi = [ "function getContractMetadata()" ];
+    let dataCall = new ethers.utils.Interface(callAbi);
```

Comment:
Struggling a bit with this part getting a 400, may have to hop on a call and work it out